code/datums/components/mood.dm

```diff
@@ -188,9 +203,87 @@
 			setSanity(sanity+0.4, SANITY_NEUTRAL, SANITY_MAXIMUM)
 		if(9)
 			setSanity(sanity+0.6, SANITY_NEUTRAL, SANITY_MAXIMUM)
+	var/psych_adjustment = ((sanity-SANITY_MAXIMUM/2)/SANITY_MAXIMUM + (mood-MOOD_LEVEL_HAPPY4/2)/MOOD_LEVEL_HAPPY4)/10
+
+	adjustPsychInstability(psych_adjustment)
+	CheckPsychInstability()
 	HandleNutrition()
 
-///Sets sanity to the specified amount and applies effects.
+
+
+#define PSYCH_BREAK_EXTREME 100
+#define PSYCH_BREAK 50 to 99
+
+#define PSYCH_CURE_EXTREME -100
+#define PSYCH_CURE -50 to -99
+
+/datum/component/mood/proc/adjustPsychInstability(amount)
+	psych_instab = CLAMP(psych_instab - amount,  PSYCH_CURE_EXTREME, PSYCH_BREAK_EXTREME)
+
+
+/datum/component/mood/proc/CheckPsychInstability()
+	switch(psych_instab)
+		if(PSYCH_BREAK_EXTREME)
+			ForceGainRandomDisorder()
+
+		if(PSYCH_BREAK)
+			RollGainRandomDisorder()
+
+		if(PSYCH_CURE)
+			RollCureRandomDisorder()
+
+		if(PSYCH_CURE_EXTREME)
+			ForceCureRandomDisorder()
+/datum/component/mood/proc/AddDisorder(datum/brain_trauma/psychological/chosen_disorder)
+	var/datum/brain_trauma/psychological/CD = chosen_disorder
+	var/mob/living/carbon/human/owner = parent
+	if(!(chosen_disorder in aquired_disorders))
+		aquired_disorders += chosen_disorder
+		owner.gain_trauma(CD , TRAUMA_RESILIENCE_ABSOLUTE)
+		adjustPsychInstability(50)
+
+/datum/component/mood/proc/CureDisorder(datum/brain_trauma/psychological/chosen_disorder)
+	var/datum/brain_trauma/psychological/CD = chosen_disorder
+	var/mob/living/carbon/human/owner = parent
+	if(chosen_disorder in aquired_disorders)
+		aquired_disorders -= chosen_disorder
+		owner.cure_trauma_type(CD, TRAUMA_RESILIENCE_ABSOLUTE)
+		adjustPsychInstability(-50)
+
+/datum/component/mood/proc/ForceGainRandomDisorder()
+	var/mob/living/carbon/human/owner = parent
+	var/datum/brain_trauma/psychological/chosen_disorder = pick(disorders - aquired_disorders)
+	if(!(chosen_disorder in aquired_disorders))
+		aquired_disorders += chosen_disorder
+		owner.gain_trauma(chosen_disorder , TRAUMA_RESILIENCE_ABSOLUTE)
+		adjustPsychInstability(50)
+
+/datum/component/mood/proc/ForceCureRandomDisorder() //not up to the code
+	var/mob/living/carbon/owner = parent
+	var/datum/brain_trauma/psychological/chosen_disorder = pick(aquired_disorders)
+	aquired_disorders -= chosen_disorder
+	owner.cure_trauma_type(chosen_disorder , TRAUMA_RESILIENCE_ABSOLUTE)
+	adjustPsychInstability(-50)
+
+/datum/component/mood/proc/RollGainRandomDisorder()
+	if(!prob(abs(psych_instab))/100)
```

Comment:
I think the not operator will take precedence here. What is the `/100` even for?